### DISCUSSION TITLE
I can't simulate a Start-Up controlling the 3 blades at the same time

### DISCUSSION THREAD

**Discussion by ACH52100** [OpenFAST/openfast#1463] [Q&A]
Hello !
I m quite bothered with my start-up simulations. To simulate a start-up, I use the test case 18 with a 5MW windturbine in wich I impose to start the active Pitch Control after 200 sec and using the PC_mode 5 in the ServoDyn module.
![image](https://user-images.githubusercontent.com/123339005/220408498-2d541027-a575-43b4-b8cc-c124d49536d6.png)
 And to set the initial conditions, I changed the ElastoDyn file by setting the 3 blades at the feathered position (90°) and a rotor speed of 0 RPM. 
![image](https://user-images.githubusercontent.com/123339005/220409645-c164d4dc-6443-49e9-b072-71125514b07f.png)
The problem is that when I launch the simulation with those parameters, FAST tells me that the speed of the blade at the tip reach 0,7 the sound velocity exactly at the moment where the active pitch control is activated.
So, I tried a start-up with only 2 blades (n°2 and 3) and the problem disappears, leading to a normal simulation, with 2 blades start-up as you can see on the following picture  
![Power_RPM_Time_Startup_11ms_2blades](https://user-images.githubusercontent.com/123339005/220410595-41a8d507-7d7a-49e1-8feb-1427d7d6ec87.png)
In fact, the problem occurs only if try to put the blade n°1 into feathered position at the beginning of the simulation and I don't know why...
Can somebody help me please ?

**Comment by jjonkman** [comment]
Dear @ACH52100,

The baseline controller for the NREL 5-MW baseline wind turbine only contains logic for normal operation.  In your case, you enable the controller at 200 s and the controller thinks the wind turbine is operating normally at this time, but in reality you want to initiate a start-up mode that does not exist.  The controller would likely get very confused why the rotor speed is near zero and and the pitch angle is near 90 degrees (which is not a normal operating condition) and would likely act oddly.

To properly implement a start-up event, I would either use the simple built-in control options available in ServoDyn for pitch ramps and generator enabling, or update the controller source code to implement your desired start-up logic.  Alternatively, you could switch to a more reasonable baseline controller, such as ROSCO: https://github.com/NREL/ROSCO, although I'm not aware that start-up control logic has been implemented in ROSCO as of yet.  (Perhaps @dzalkind can correct me.)

Best regards,

**Reply by dzalkind** [reply]
Hi,

ROSCO does not have start-up control, though it is planned for the future.  ROSCO does have open-loop control that could be adapted into start-up control.  If you're interested in that, we should move the discussion to ROSCO.  

Simulink is also a nice way to prototype control solutions.

Best, Dan

**Reply by ACH52100** [reply]
Thank you very much for your point of vue. I understand better why the simulation can't be runned with the Active Pitch control.
As you suggested, I also tried to simulate a startup using the simple built-in control options available in ServoDyn. And I succeed to run the simulation with a 3 blades normal start-up. But I still have a problem with the configuration of the generator. The blades are correctly managed, going from 90° to the normal pitch angle according to the wind speed (as example below, with a wind speed of 20 m/s, the pitch reached the required pitch angle to ensure an output power of 5MW).
![Momentum_Angle_Time_StartupManual_20ms](https://user-images.githubusercontent.com/123339005/220439695-bbe4b809-6125-44ff-b600-d2b59d63ed22.png) 
![Power_RPM_Time_StartupManual_20ms](https://user-images.githubusercontent.com/123339005/220439757-b5c46e51-a3ad-4e24-b4d4-1f4c5ba7f1a6.png)
But I have a lot of noise at the beginning for several wind speed, the wind turbine can't even start like below, with a wind speed of 15 m/s
![Power_RPM_Time_StartupManual_15ms](https://user-images.githubusercontent.com/123339005/220440415-ff72aac1-1caa-4d91-a849-d21da04d9536.png)
I think this is due to my generator parameters, but I don't know how to manage it, even with the Fast User's Guide. My parameters are below:
![image](https://user-images.githubusercontent.com/123339005/220441790-50864e89-0e66-4f3a-b5f5-0514c3fcf741.png)
I wanted to simulate the start-up using "SpdGenOn". But I don't know what value would be appropriated. I tried with different values, without success. Can you tell me what value I should set for "SpdGenOn" to simulate my start-up please ?
Thank you very much

**Reply by jjonkman** [reply]
Dear @ACH52100,

I would set SpdGenOn to the minimum speed of the generator when power is produced, which is 670 rpm for the NREL 5-MW baseline wind turbine.

Please note, however, that override pitch maneuvers, which you are using to command the pitch from feather to run for start up, will also override the commands from the pitch controller, once engaged.  So, the approach you are using likely wont work for start-up events at high speeds (above rated, where the pitch controller is active).  This is likely why you are seeing oscillations in speed and power.  This is where implementation of the start-up logic in the actually controller would be preferred.

Best regards,
